### PR TITLE
Prepare templates repository for v2.0.0 launch

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #107
Fixes: #104 - please note that this only for CSharp and Go and excludes FSharp and VB
Fixes: https://github.com/pulumi/pulumi/issues/4216
 
This also adds a kubernetes-go template

Related: https://github.com/pulumi/home/issues/746

The reason this is in draft is that we want to have the v2.0.0 dependency in the go.mod for the testing.go file - this can't be done until the release has been cut

### CODE REVIEW

**Comment by justinvp** [comment]
Also bump `"@types/node": "^8.0.0"` to `"@types/node": "^10.0.0"` given https://github.com/pulumi/docs/pull/2724?

**Comment by stack72** [comment]
> Also bump `"@types/node": "^8.0.0"` to `"@types/node": "^10.0.0"` given [pulumi/docs#2724](https://github.com/pulumi/docs/pull/2724)?

Any reason we would choose 10.0.0 and not go further up the list to 13.x.x?

**Comment by justinvp** [comment]
> Any reason we would choose 10.0.0 and not go further up the list to 13.x.x?

10 is the minimum Active release we support and is what @lukehoban chose [here](https://github.com/pulumi/docs/pull/2724/files#diff-03325a6302d74f8cc58805a36a1793ec):

```diff
-        "@types/node": "^8.0.26"
+        "@types/node": "^10.0.0"
```

I'll defer to him.

**Comment by lukehoban** [comment]
Yeah - sticking with `^10.0.0` makes sense to me for now.  That is the minimum we support, and so will pick up type definitions that are guaranteed to work for any Pulumi user.  Folks who are on newer versions can manually update this to depend on more recent Node.js features (though that is fairly rare).

File: alicloud-csharp/${PROJECT}.csproj [hunk 1]
```diff
@@ -7,7 +7,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Pulumi.Alicloud" Version="1.5.0-preview" />
+    <PackageReference Include="Pulumi.Alicloud" Version="2.0.0" />
```
Comment (mikhailshilkov) [inline alicloud-csharp/${PROJECT}.csproj:10]: I think we want to use floating versions like `2.*` to avoid changing templates frequently

**Comment by stack72** [comment]
@justinvp / @lukehoban / @mikhailshilkov feedback acted upon

**Pull request merged**